src/compiler/program.ts

```diff
@@ -436,6 +436,8 @@ namespace ts {
         host = host || createCompilerHost(options);
 
         let skipDefaultLib = options.noLib;
+        const getDefaultLibraryFileName = memoize(() => host.getDefaultLibFileName(options));

```

Comment:
Why bother with `memoize` and not just have `const defaultLibraryFileName = host.getDefaultLibraryFileName(options)`?

Reply:
Depending on the options and some logic, the variable might end up not being needed. This was done for lazy evaluation.